kalite/topic_tools/content_recommendation.py

```diff
@@ -89,25 +56,59 @@ def get_resume_recommendations(user):
 # - Topic tree structure recommendations based on the most recent subtopic accessed
 #
 # @param user: facility user model
-#        current_subtopics: subtopic ids of the 5 most recently accessed exercise
-# @return: a list of exercise id's of where the user should consider going next.
+#        
+# @return: a list of exercise id's and their titles, [type], and subtopic for where the user 
+#          should consider going next.
 ###
 def get_next_recommendations(user):
 
+    #user = ExerciseLog.objects.filter(id__lte=1)[4].user        #random person, can delete after    
+
+    exercise_parents_table = get_exercise_parents_lookup_table()
+    topic_table = get_topic_tree_lookup_table()
+
+    most_recent = get_most_recent_exercises(user)
+
+    if len(most_recent) > 0 and most_recent[0] in exercise_parents_table:
+        current_subtopic = exercise_parents_table[most_recent[0]]['subtopic_id']
+    else:
+        current_subtopic = None
+
     #logic for recommendations based off of the topic tree structure
-    current_subtopic = get_most_recent_subtopic(user)                   #the most recent subtopic id accessed by user
-    topic_tree_based_data = get_recommended_exercises(current_subtopic)
-   
+    if current_subtopic:
+        topic_tree_based_data = generate_recommendation_data()[current_subtopic]['related_subtopics'][:3]
```

Comment:
Cool, we don't have to generalize it now -- just move the number 3 here to a global variable/setting, and then file an issue for it so that we don't forget.